packages/beacon-state-transition/src/allForks/util/epochContext.ts

```diff
@@ -329,6 +331,41 @@ export class EpochContext {
     return validatorIndices;
   }
 
+  /**
+   * @param validatorData pubkeys indexed by validator index
+   */
+  getCommitteeAssignments(epoch: Epoch, validatorData: Map<number, BLSPubkey>): routes.validator.AttesterDuty[] {
```

Comment:
Change to 
```
getCommitteeAssignments(epoch: Epoch, requestedValidatorIndices: Set<number>): AttesterDuty[]
```
and use the set to validate if index belongs in set instead of map. The consumer of this function may add the pubkey if necessary.